### PR TITLE
Event emitter provider

### PR DESCRIPTION
**Pull request by jack-berg**
Split out the event API so that it is completely separate from the log API, rather than dependent on the log API as in #4961.

Summary of changes:
- Introduce new `:api:events` module, containing:
  - EventEmitterProvider (and GlobalEventEmitterProvider)
  - EventEmitterBuilder
  - EventEmitter
- Remove event API surface area from `:api:logs`
- Refactor `:sdk:logs` to implement both log API and event API. I.e.:
  - `SdkLoggerProvider` implements both `LoggerProvider` and `EventEmitterProvider`
  - `SdkLoggerBuilder` implements both `LoggerBuilder` and `EventEmitterBuilder`
  - `SdkLogger` implements both `Logger` and `EventEmitter`
- Update autoconfigure to call `GlobalEventEmitterProvider.set`

Draft for now as this is _not_ what is currently described in the [Event API spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/event-api.md). This approach is suggested by @trask and @jkwatson [here](https://github.com/open-telemetry/opentelemetry-specification/pull/2966#issuecomment-1353489049), and by myself a while ago in the [otep](https://github.com/open-telemetry/oteps/pull/202#discussion_r880902181).

### CODE REVIEW

File: api/events/src/main/java/io/opentelemetry/api/events/EventEmitter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A {@link EventEmitter} is the entry point into an event pipeline.
+ *
+ * <p>Example usage emitting events:
+ *
+ * <pre>{@code
+ * class MyClass {
+ *   private final EventEmitter eventEmitter = openTelemetryEventEmitterProvider.loggerBuilder("instrumentation-library-name", "acme.observability")
```
Comment (jkwatson) [inline api/events/src/main/java/io/opentelemetry/api/events/EventEmitter.java:18]: something doesn't look right on this line.... loggerBuilder?

File: api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+   * Gets or creates a named EventEmitter instance which emits events to the {@code eventDomain}.
+   *
+   * @param instrumentationScopeName A name uniquely identifying the instrumentation scope, such as
+   *     the instrumentation library, package, or fully qualified class name. Must not be null.
+   * @param eventDomain The event domain, which acts as a namespace for event names. Within a
+   *     particular event domain, event name defines a particular class or type of event.
+   * @return a Logger instance.
+   */
+  default EventEmitter get(String instrumentationScopeName, String eventDomain) {
```
Comment (jkwatson) [inline api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java:28]: Is eventDomain mandatory? I don't even know what sort of values I'd put in for this.

Comment (jack-berg) [reply api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java:28]: As of now `event.domain` is [required](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/semantic_conventions/events.md). The spec describes it as:
> The event.domain attribute is used to logically separate events from different systems. For example, to record Events from browser apps, mobile apps and Kubernetes, we could use browser, device and k8s as the domain for their Events. This provides a clean separation of semantics for events in each of the domains. Within a particular domain, the event.name attribute identifies the event. Events with same domain and name are structurally similar to one another.

There's an open [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/2994) to discuss whether event.domain and event.name can be merged into a single field.

**Comment by tigrannajaryan** [comment]
@jack-berg do I understand correctly that with this PR we end with roughly the following modules and data flows:
image

Question: do we need the "Log API" in the API package or we can get rid of it and make Bridges/Appenders call directly into Log SDK package?

**Comment by jkwatson** [comment]
> Question: do we need the "Log API" in the API package or we can get rid of it and make Bridges/Appenders call directly into Log SDK package?

This was actually our first approach (and the one I strongly favored). Unfortunately, getting this to work with the OTel javaagent ended up being quite problematic (@trask can probably explain the details, since it's pretty fuzzy in my head), leading us to create an API surface for the javaagent to interact with.

**Comment by jack-berg** [comment]
> @jack-berg do I understand correctly that with this PR we end up with roughly the following modules and data flows:

Yes that's correct.

> Question: do we need the "Log API" in the API package or we can get rid of it and make Bridges/Appenders call directly into Log SDK package?

@tigrannajaryan I responded to the question of whether or not we need a log appender API [here](https://github.com/open-telemetry/opentelemetry-specification/issues/3086#issuecomment-1377433469). In summary, yes, we do need a log appender API.

File: api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java (hunk 2 above)
Comment (jkwatson) [reply api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java:28]: This is the only sticking point I have with this API. I don't love having to explain to people what this is supposed to be used for. We have had enough trouble explaining what the "name" of a Tracer is...this adds an additional layer of having to explain this arcane idea that almost no one will care about. Can we make it optional? Or default to something like "none" or "unknown" or "who cares?"? :)

Comment (jack-berg) [reply api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java:28]: I'm not opposed to making it optional with a default given the experimental state of the spec. The spec could reaffirm the importance and requirement of this field, in which case we'd have to come back and adjust our API, but that's fine and won't be the last breaking change to the event API. Will push a commit.

Comment (jack-berg) [reply api/events/src/main/java/io/opentelemetry/api/events/EventEmitterProvider.java]: Pushed a commit where event domain is now optional, set via `eventEmitterProvider.eventEmitterBuilder("scope-name").setEventDomain("event-domain").build()`.

If unset, the `event.domain` attribute defaults to `unknown`.

**Review by jkwatson — APPROVED** [review]
🚢

**Pull request merged**